readalongs/web_api.py

```diff
@@ -21,18 +21,23 @@
 
 import io
 import os
+from enum import Enum
+from tempfile import TemporaryDirectory
+from textwrap import dedent
 from typing import Dict, List, Optional, Union
 
 from fastapi import Body, FastAPI, HTTPException
 from fastapi.middleware.cors import CORSMiddleware
+from fastapi.responses import Response
```

Comment:
why not `FileResponse`?

Reply:
I think it's because the file has to be deleted immediately after sending it, and there isn't a way (or at least not an obvious way) to do that with FileResponse.

Reply:
It should be the same no? The response (whether `Response` or `FileResponse`) should return the file created in the temporary directory. Once it does that, the file gets deleted.

Reply:
The problem is that `FileResponse` sends it asynchronously, so the directory will get deleted (from exiting the with block) before the event loop gets around to sending it.

Reply:
I see - in that case I guess we could just write a background task to delete the file after the response is sent https://fastapi.tiangolo.com/tutorial/background-tasks/

Reply:
I first tried `FileResponse`, but it's broken for our use case: it asynchronously sends the file *after* the endpoint function exists, but I delete the temporary folder so the file is gone by the time `FileResponse` tries to read it and I just get a file-not-found error. David suggested `StreamingResponse` as an alternative, but it suffers from the same problem.

Because our data privacy policy promises to delete the file right away, I have to slurp the file into memory and send that. I dug around to find a "FileResponse from memory" solution and what I found was to put it in a plain `Response` with the headers I set.

Reply:
Interesting. I was worried about creating a race condition, but FileResponse accepts a `background` parameter provide a background task to run after running. See https://stackoverflow.com/a/68579536
This looks like a very elegant solution, I'll give it a shot.

Reply:
Dang I hate Windows! I have it all working with a FileResponse now, but if I run the tests on Windows, the SRT, VTT and EAF files get CRLF newlines instead of LF. I guess that's valid too, but what a nuisance! It means my unit testing has to convert before testing for equality. Bleh.

My slurp_file solution was reading the file back on windows, converting it to LF in the process, but a FileResponse sends the file contents verbatim.